velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -752,24 +760,37 @@ uint64_t Writer::MemoryReclaimer::reclaim(
     ++stats.numNonReclaimableAttempts;
     return 0;
   }
-  const uint64_t memoryUsage = writer_->getContext().getTotalMemoryUsage();
-  if (memoryUsage < writer_->spillConfig_->writerFlushThresholdSize) {
-    RECORD_METRIC_VALUE(kMetricMemoryNonReclaimableCount);
-    LOG(WARNING)
-        << "Can't reclaim memory from dwrf writer pool " << pool->name()
-        << " which doesn't have sufficient memory to flush, writer memory usage: "
-        << succinctBytes(memoryUsage) << ", writer flush memory threshold: "
-        << succinctBytes(writer_->spillConfig_->writerFlushThresholdSize);
-    ++stats.numNonReclaimableAttempts;
-    return 0;
-  }
 
   return memory::MemoryReclaimer::run(
       [&]() {
         int64_t reclaimedBytes{0};
         {
           memory::ScopedReclaimedBytesRecorder recorder(pool, &reclaimedBytes);
-          writer_->flushInternal(false);
+          const auto& context = writer_->getContext();
+          const auto usedBytes = context.getTotalMemoryUsage();
+          const auto releasableBytes = context.releasableMemoryReservation();
```

Comment:
Maybe simplify as
```
writer_->releaseMemory();
const auto usedBytes = context.getTotalMemoryUsage();
if (usedBytes < writer_->spillConfig_->writerFlushThresholdSize) {
  ...
} else {
  flush
}
```

Reply:
we don't need to make releaseMemory() call if there is nothing releasable.

Reply:
This is pretty cheap operation and let's keep it simple first? Thanks!